test/Worker.Extensions.Tests/ServiceBus/ServiceBusSessionMessageActionsTests.cs

```diff
@@ -49,6 +50,55 @@ public async Task CanRenewSessionLock()
             await messageActions.RenewSessionLockAsync();
         }
 
+        [Fact]
+        public async Task CanSetAndGetSessionStateWithNull()
+        {
+            var sessionId = "test";
+            var initialTestString = "TestInitialState";
+            ByteString currentSessionState = ByteString.CopyFromUtf8(initialTestString);
+
+            var mockClient = new Mock<Settlement.SettlementClient>();
+            
+            mockClient
+                .Setup(x => x.GetSessionStateAsync(
+                    It.Is<GetSessionStateRequest>(r => r.SessionId == sessionId),
+                    It.IsAny<Metadata>(),
+                    It.IsAny<DateTime?>(),
+                    It.IsAny<CancellationToken>())
+                )
+                .Returns(() => CreateAsyncUnaryCall(new GetSessionStateResponse
+                {
+                    SessionState = currentSessionState
+                }));
+
+            mockClient
+                .Setup(x => x.SetSessionStateAsync(
+                    It.Is<SetSessionStateRequest>(r => r.SessionId == sessionId),
+                    It.IsAny<Metadata>(),
+                    It.IsAny<DateTime?>(),
+                    It.IsAny<CancellationToken>())
+                )
+                .Returns<SetSessionStateRequest, Metadata, DateTime?, CancellationToken>((request, _, _, _) =>
+                {
+                    currentSessionState = request.SessionState;
```

Comment:
For the test verifying the return on `SetSessionStateAsync`, you shouldn't need to capture this parameter in a lambda. Instead, you use a mock verify call to assert how it was called.

``` csharp
mockClient.Verify(m => m.SetSessionStateAsync(<It.* statements to verify precisely how it should be called>), Times.Once());
```